pkg/sync/sync_context.go

```diff
@@ -934,6 +934,20 @@ func (sc *syncContext) applyObject(t *syncTask, dryRun, force, validate bool) (c
 			message, err = sc.resourceOps.CreateResource(context.TODO(), t.targetObj, dryRunStrategy, validate)
 		}
 	} else {
+		targetObjIsSSA := resourceutil.HasAnnotationOption(t.targetObj, common.AnnotationSyncOptions, common.SyncOptionServerSideApply)
+		liveObjIsNotSSA := t.liveObj != nil && !resourceutil.HasAnnotationOption(t.liveObj, common.AnnotationSyncOptions, common.SyncOptionServerSideApply)
+		if t.targetObj.GetKind() == kube.NamespaceKind && t.liveObj != nil && t.liveObj.GetKind() == kube.NamespaceKind && targetObjIsSSA && liveObjIsNotSSA {
+			// If there is a pre-existing live namespace, and if we also want to use managedNamespaceMetadata on an existing namespace,
+			// we need to do `kubectl apply` with the namespace as-is, using a different manager than the default one.
+			// The reason we do this is so that we can preserve existing labels and annotations on the namespace.
+			nsSpec := &v1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: kube.NamespaceKind}, ObjectMeta: metav1.ObjectMeta{Name: t.liveObj.GetName(), Annotations: t.liveObj.GetAnnotations(), Labels: t.liveObj.GetLabels(), Namespace: t.liveObj.GetNamespace()}}
+			liveCopy, err := kube.ToUnstructured(nsSpec)
+
+			message, err = sc.resourceOps.ApplyResource(context.TODO(), liveCopy, dryRunStrategy, force, validate, serverSideApply, "argocd-controller-tmp")
```

Comment:
`argocd-controller-tmp` is a placeholder name, but the idea is that in order to preserve existing metadata we need to apply the namespace with a separate manager before using `managedNamespaceMetadata` on said namespace.